src/Apps/Order/Utils/formValidators.tsx

```diff
@@ -8,6 +8,15 @@ export const validatePresence = (value: string): string | null => {
   return null
 }
 
+export const validatePostalCode = (postalCode: string, countryCode: string) => {
+  let postalCodeRegex: RegExp =
+    countryCode === "US"
+      ? /^([0-9]{5})(?:[-\s]*([0-9]{4}))?$/
+      : /^([A-Z][0-9][A-Z])\s*([0-9][A-Z][0-9])$/
+
+  return postalCodeRegex.test(postalCode) ? null : "This field is required"
```

Comment:
**suggestion:** If a postal code is provided, but does not match the regex, this error message won't make sense to the user. I would suggest keeping `validatePresence(postalCode)` in the `postalCode` validation check below.

Reply:
makes sense. that way we can return/display different messages to the user for empty and wrong postal code. thank you!